.github/workflows/prChecks.yml

```diff
@@ -0,0 +1,46 @@
+name: Build
+run-name: Checking the build after the Push/PR by ${{ github.actor }}
+on: [push]
+jobs:
+  Build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          build-scan-publish: true
+          build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
+          build-scan-terms-of-use-agree: "yes"
+
+      - name: Build with Gradle
+        run: ./gradlew build
+
+      - name: Publish Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v2.16.1
+        if: always()
+        with:
+          files: |
+            test-results/**/*.xml
+            test-results/**/*.trx
+            test-results/**/*.json
+
+      - name: TestForest Dashboard
+        uses: test-summary/action@v2.3
+        with:
+          paths: "test/results/**/TEST-*.xml"
+          if: always()
```

Comment:
yeah, i can't figure out what the path should be, its `build/test-results/test/TEST-actions.APPTest.xml`

you are right i just needed one but i was trying to add summary and a bit more context to it. But nothing is working right now so can't see how they look

Reply:
you can use the pattern `build/test-results/test/TEST-*.xml`

.json and .trx is not needed here as our setup will only write .xml files

Reply:
`build` is the output directory of gradle. So all generated files will end up there.

`test-results/test` is the location used by the gradle test task that runs the tests

`TEST-*.xml` is the output file of the test reporter which will use .XML (compare https://docs.gradle.org/current/userguide/java_testing.html#test_reporting)

Reply:
if i take out `build`, it doesn't work. it worked in the last run where i had the `build` in path. so not sure what is going on 🤷‍♂️

Reply:
ran it again with the `build/` path and it works again 
https://github.com/muhammad83/doThis/actions/runs/9528555635

Reply:
we now have the test results available in 2 locations:
PR comment: https://github.com/muhammad83/doThis/pull/7#issuecomment-2168683757
GitHub check: https://github.com/muhammad83/doThis/pull/7/checks?check_run_id=26266368872

I guess this is done by the both plugins but not sure which one does what?
Feel free to leave in both or decide on one. i leave that up to you.

Looking good now! Will approve

Reply:
> if i take out build, it doesn't work. it worked in the last run where i had the build in path. 

you have to use the build prefix. To reproduce locally use these steps:

```
$ cd doThis
$ gw build
$ ls build/test-results/test/TEST-*.xml
build/test-results/test/TEST-com.softbits.doThis.ApplicationTests.xml
```

We build the project (this includes running the tests and creating the report)
With the `ls` command you can verify the glob pattern. If you use one of the old patterns you'll see that they won't match anything

```
❯ ls test-results/**/*.xml
fish: No matches for wildcard 'test-results/**/*.xml'. See `help wildcards-globbing`.
ls test-results/**/*.xml
   ^~~~~~~~~~~~~~~~~~~~^
```

Reply:
but don't worry. debugging github actions sometimes is a pita because of the longer feedback loop.

You need to push a change to the repo and then wait for the result. This is more mental load than working locally and e.g. getting some instant feedback from the compiler or a unit test

Reply:
thank you for the explanation, i didn't know these commands. now i do, thank you.